File: data/scripts/ini4.py
filename = "dataset.txt"
a, b = 0, 0
with open(filename) as f:
    for riga in f:
           lista = riga.split(" ") 
           a, b = int(lista[0]), int(lista[1])
           break   
tot = 0
for n in range(a, b+1):
	   if n%2 == 1:
	   	  tot+=n
print(tot)	   	  

File: public/js-solutions/ini4.mjs
// ini4 - esercizio introduttivo "Python Village": dati due interi a e b
// (a <= b) sulla stessa riga, somma tutti gli interi dispari nell'
// intervallo [a, b]. Soluzione JavaScript indipendente, non una
// trascrizione di problem.py, riscritta in modo idiomatico per JS.
//
// Contratto: riceve il contenuto testuale di dataset.txt, restituisce
// l'output testuale (stessa forma dell'output Python).
export default function solve(datasetText) {
  const prima_riga = datasetText.split("\n")[0].trim();
  const parti = prima_riga.split(/\s+/);

  if (parti.length !== 2 || parti.some((p) => !/^-?\d+$/.test(p))) {
    throw new Error(`Input non valido: attesi due interi "a b", ricevuto "${prima_riga}"`);
  }

  const [a, b] = parti.map(Number);

  let tot = 0;
  for (let n = a; n <= b; n++) {
    if (n % 2 !== 0) tot += n;
  }

  return `${tot}\n`;
}
